tests/lib/rules/no-array-constructor.js

```diff
@@ -525,113 +725,73 @@ ruleTesterTypeScript.run("no-array-constructor", rule, {
 	invalid: [
 		{
 			code: "new Array();",
+			output: "[];",
 			errors: [
 				{
-					messageId: "preferLiteral",
-					suggestions: [
-						{
-							messageId: "useLiteral",
-							output: "[];",
-						},
-					],
+					messageId: "useLiteral",
 				},
 			],
 		},
 		{
 			code: "Array();",
+			output: "[];",
 			errors: [
 				{
-					messageId: "preferLiteral",
-					suggestions: [
-						{
-							messageId: "useLiteral",
-							output: "[];",
-						},
-					],
+					messageId: "useLiteral",
 				},
 			],
 		},
 		{
 			code: "new Array(x, y);",
+			output: "[x, y];",
 			errors: [
 				{
-					messageId: "preferLiteral",
-					suggestions: [
-						{
-							messageId: "useLiteral",
-							output: "[x, y];",
-						},
-					],
+					messageId: "useLiteral",
 				},
 			],
 		},
 		{
 			code: "Array(x, y);",
+			output: "[x, y];",
 			errors: [
 				{
-					messageId: "preferLiteral",
-					suggestions: [
-						{
-							messageId: "useLiteral",
-							output: "[x, y];",
-						},
-					],
+					messageId: "useLiteral",
 				},
 			],
 		},
 		{
 			code: "new Array(0, 1, 2);",
+			output: "[0, 1, 2];",
 			errors: [
 				{
-					messageId: "preferLiteral",
-					suggestions: [
-						{
-							messageId: "useLiteral",
-							output: "[0, 1, 2];",
-						},
-					],
+					messageId: "useLiteral",
 				},
 			],
 		},
 		{
 			code: "Array(0, 1, 2);",
+			output: "[0, 1, 2];",
 			errors: [
 				{
-					messageId: "preferLiteral",
-					suggestions: [
-						{
-							messageId: "useLiteral",
-							output: "[0, 1, 2];",
-						},
-					],
+					messageId: "useLiteral",
 				},
 			],
 		},
 		{
 			code: "Array?.(0, 1, 2);",
+			output: "[0, 1, 2];",
 			errors: [
 				{
-					messageId: "preferLiteral",
-					suggestions: [
-						{
-							messageId: "useLiteral",
-							output: "[0, 1, 2];",
-						},
-					],
+					messageId: "useLiteral",
 				},
 			],
 		},
 		{
 			code: "Array?.(x, y);",
+			output: "[x, y];",
```

Comment:
We should only autofix when `Array` refers to the global built-in.
If Array is shadowed locally, applying the fix could cause unintended behavior:

```js
function myScope() {
  const Array = undefined;
  Array?.(1, 2);
}
```

Although this is a rare case, WDYT?

Reply:
The rule already checks if `Array` refers to the global built-in:

https://github.com/eslint/eslint/blob/e62e26761561e1d78c6466a2d74dbf946012fddc/lib/rules/no-array-constructor.js#L96-L105

And in that case, it doesn't report a problem: [Playground demo](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWFycmF5LWNvbnN0cnVjdG9yOiBcImVycm9yXCIgKi9cblxuZnVuY3Rpb24gbXlTY29wZSgpIHtcbiAgY29uc3QgQXJyYXkgPSB1bmRlZmluZWQ7XG4gIEFycmF5Py4oMSwgMik7XG59Iiwib3B0aW9ucyI6eyJydWxlcyI6e30sImxhbmd1YWdlT3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319LCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYVZlcnNpb24iOiJsYXRlc3QifX19)